src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,180 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            addFirst(value);
+        } else {
+            addAfter(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (isEmpty() && index == 0) {
+            addFirst(value);
+        } else if (index == 0) {
+            addBefore(value);
+        } else if (index == size) {
+            addAfter(value);
+        } else {
+            link(nodeOf(index), value);
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return nodeOf(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        verifyIndexIsInRange(index);
```

Comment:
`verifyIndexIsInRange(index);` is redundant, it is already present inside of `nodeOf()` method
please check the other same code occurrences.